### PR TITLE
Sidm 2686 cross browser tests nightly jenkins

### PR DESCRIPTION
**Pull request by sudhasane**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

https://tools.hmcts.net/jira/browse/SIDM-2686

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[N ] No
```

### CODE REVIEW

File: src/test/js/policy_check_functional_test.js [hunk 1]
```diff
@@ -28,10 +29,10 @@ BeforeSuite(async (I) => {
...
     const serviceRoles = [serviceBetaRole, serviceAdminRole, serviceSuperRole];
     roleNames.push(serviceRoles);
-    await I.createServiceWithRoles(serviceName, serviceRoles, serviceBetaRole, token);
-    serviceNames.push(serviceName);
-    await I.createUserWithRoles(adminEmail, randomUserFirstName + 'Admin', [serviceAdminRole, "IDAM_ADMIN_USER"]);
-    userFirstNames.push(randomUserFirstName + 'Admin');
+      await I.createServiceWithRoles(serviceName, serviceRoles, serviceBetaRole, token);
+      serviceNames.push(serviceName);
+//    await I.createUserWithRoles(adminEmail, randomUserFirstName + 'Admin', [serviceAdminRole, "IDAM_ADMIN_USER"]);
+//    userFirstNames.push(randomUserFirstName + 'Admin');
```
Comment (tbamido) [inline src/test/js/policy_check_functional_test.js]: are these not needed then?

File: src/test/js/shared/custom_steps.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ module.exports = function() {
         this.dontSeeInCurrentUrl('attemptfive');
         this.fillField('username', email);
         this.fillField('password', '111111111111111111');
-        this.scrollPageToBottom();
+        //XB: this.scrollPageToBottom();
```
Comment (tbamido) [inline src/test/js/shared/custom_steps.js]: what does XB mean?

**Comment by sudhasane** [comment]
> left some questions regarding the JavaScript bits
> can't really say much on the Devops bits... lots of complex changes need some devops check

Devops changes were not done by me, current branch (Sidm 2686 cross browser tests nightly jenkins)has been created from saucelabs-branch (where code for running cross browser tests was written, this is quite old branch) , as I pulled the latest code from Preview  it's been updated with new changes

File: src/test/js/policy_check_functional_test.js (hunk 1 above)
Comment (sudhasane) [reply src/test/js/policy_check_functional_test.js]: Yes, for this test run not required

File: src/test/js/shared/custom_steps.js (hunk 2 above)
Comment (sudhasane) [reply src/test/js/shared/custom_steps.js]: I am not sure about this ,this was there in old branch where sauce labs code was written, I tried rerunning test on sauce labs by using existing code in preview branch, which is working fine , So I am removing this and replacing with code in preview branch

**Comment by henrydobson** [comment]
@sudhasane This PR is based on master. It cannot be. Please update to preview.

File: src/test/js/reset_password_test.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ AfterSuite(async (I) => {
     return await I.deleteAllTestData(randomData.TEST_BASE_PREFIX);
 });
 
-Scenario('@functional @resetpass As a citizen user I can reset my password', async (I) => {
+Scenario('@functional  @resetpass As a citizen user I can reset my password', async (I) => {
```
Comment (henrydobson) [inline src/test/js/reset_password_test.js]: no change here except whitespace. Revert this change so that there's no diff

File: src/test/js/policy_check_functional_test.js [hunk 4]
```diff
@@ -45,7 +44,7 @@ AfterSuite(async (I) => {
     ]);
 });
 
-Scenario('@functional @policy As a citizen with policies blocking me from login I should see an error message', async (I) => {
+Scenario('@functional  @policy As a citizen with policies blocking me from login I should see an error message', async (I) => {
```
Comment (henrydobson) [inline src/test/js/policy_check_functional_test.js]: no change here except whitespace. Revert this change so that there's no diff

File: src/test/js/policy_check_functional_test.js [hunk 5]
```diff
@@ -28,10 +29,8 @@ BeforeSuite(async (I) => {
     const serviceSuperRole = response.name;
     const serviceRoles = [serviceBetaRole, serviceAdminRole, serviceSuperRole];
     roleNames.push(serviceRoles);
-    await I.createServiceWithRoles(serviceName, serviceRoles, serviceBetaRole, token);
-    serviceNames.push(serviceName);
-    await I.createUserWithRoles(adminEmail, randomUserFirstName + 'Admin', [serviceAdminRole, "IDAM_ADMIN_USER"]);
-    userFirstNames.push(randomUserFirstName + 'Admin');
+      await I.createServiceWithRoles(serviceName, serviceRoles, serviceBetaRole, token);
```
Comment (henrydobson) [inline src/test/js/policy_check_functional_test.js]: Remove extra indentation.

File: src/test/js/cross_browser_test.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    I.wait(2)
+    I.fillField('#password', TestData.PASSWORD);
+    I.click('Sign in');
+});
+
+
+
+
+
+
```
Comment (henrydobson) [inline src/test/js/cross_browser_test.js]: Dirty file ending

File: src/test/js/config/test_data.js [hunk 7]
```diff
@@ -6,6 +6,6 @@ module.exports = {
     NOTIFY_API_KEY: process.env.NOTIFY_API_KEY,
     SCENARIO_RETRY_LIMIT: 3,
     PASSWORD: "Passw0rdIDAM",
```
Comment (henrydobson) [inline src/test/js/config/test_data.js]: All secrets should be fetched from keyvault. Do not hardcode.

File: src/test/js/config/runUITestsSauceLabs.sh [hunk 8]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+ supportedBrowsers=`sed '/\/\//d' ./supportedBrowsers.js | sed '/: {/!d' | sed "s/[\'\:\{ ]//g"`
```
Comment (henrydobson) [inline src/test/js/config/runUITestsSauceLabs.sh]: Remove leading whitespace

Comment (henrydobson) [reply src/test/js/config/runUITestsSauceLabs.sh]: back ticks are not advised for this kind of assignment. Use `$(sed '/\/\//d' ...)` instead.

File: saucelabs.conf.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    safari: {
+      browsers: getBrowserConfig('safari')
+    }
+  },
+  name: 'Idam web public'
+};
+
+exports.config = setupConfig;
+
```
Comment (henrydobson) [inline saucelabs.conf.js]: Dirty file ending

File: src/test/js/config/runUITestsSauceLabs.sh [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+    for f in ../../../../output/*.*; do
+        echo $f
+        mv $f ../../../../output/$FOLDERNAME
+    done
+
+    exitStatus=$?
+    if [ $exitStatus -ne 0 ]; then
+        finalExitStatus=$exitStatus
+    fi
```
Comment (henrydobson) [inline src/test/js/config/runUITestsSauceLabs.sh]: This looks like it does nothing. Why is it here?

File: src/test/js/policy_check_functional_test.js [hunk 11]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (henrydobson) [inline src/test/js/policy_check_functional_test.js]: Remove leading new line

File: yarn.lock [hunk 12]
```diff
@@ -2,205 +2,332 @@
 # yarn lockfile v1
```
Comment (henrydobson) [inline yarn.lock]: This can be removed from git and gitignored

**Comment by henrydobson** [comment]
@sudhasane This PR need refinement. It should not have been branched from such an old branch really and there are many quality issues. I'd like to see a link to a passing build using the nightly Jenkinsfile but before that, the functionality of the nightly pipeline is not as desired. This pipeline file will only all tests against one environment despite the parameters for the test urls. Please address these issues and ensure that the pipeline can target any environment and defaults to aat.

File: src/test/js/config/runUITestsSauceLabs.sh (hunk 10 above)
Comment (sudhasane) [reply src/test/js/config/runUITestsSauceLabs.sh]: This is not written by me, it was there in the branch from which I branched and was written by previous Amido QA

File: src/test/js/config/runUITestsSauceLabs.sh (hunk 8 above)
Comment (sudhasane) [reply src/test/js/config/runUITestsSauceLabs.sh]: This is also not written by me, but I will apply the changes as suggested

File: yarn.lock (hunk 12 above)
Comment (sudhasane) [reply yarn.lock]: > @sudhasane This PR need refinement. It should not have been branched from such an old branch really and there are many quality issues. I'd like to see a link to a passing build using the nightly Jenkinsfile but before that, the functionality of the nightly pipeline is not as desired. This pipeline file will only all tests against one environment despite the parameters for the test urls. Please address these issues and ensure that the pipeline can target any environment and defaults to aat.

we run nightly tests only on AAT thats the requirement from hmcts

File: Jenkinsfile_nightly [hunk 13]
```diff
@@ -2,47 +2,44 @@
...
     pipelineTriggers([cron('10 21 * * *')]),
-
     parameters([
-
-        string(name: 'URL_TO_TEST', defaultValue: 'https://idam-web-public-idam-preview.service.core-compute-idam-preview.internal', description: 'The URL you want to run these tests against'),
-        string(name: 'API_URL_TO_TEST', defaultValue: 'https://idam-api-idam-preview.service.core-compute-idam-preview.internal', description: 'The API URL you want to run these tests against '),
+        string(name: 'URL_TO_TEST', defaultValue: 'https://idam-web-public-idam-aat.service.core-compute-idam-aat.internal', description: 'The URL you want to run these tests against'),
+        string(name: 'API_URL_TO_TEST', defaultValue: 'https://idam-api-idam-aat.service.core-compute-idam-aat.internal', description: 'The API URL you want to run these tests against '),
+        string(name: 'SAUCE_KEY', defaultValue: '101a14e3-0458-4927-a968-129adbf18152', description: 'Sauce key'),
+        string(name: 'SAUCE_USERNAME', defaultValue: 'sudhasane', description: 'Sauce User name'),
```
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: as previously mentioned its should run on AAT which is master

> @sudhasane This PR need refinement. It should not have been branched from such an old branch really and there are many quality issues. I'd like to see a link to a passing build using the nightly Jenkinsfile but before that, the functionality of the nightly pipeline is not as desired. This pipeline file will only all tests against one environment despite the parameters for the test urls. Please address these issues and ensure that the pipeline can target any environment and defaults to aat.

we run nightly tests only on AAT thats the requirement from hmcts

File: Jenkinsfile_nightly [hunk 14]
```diff
@@ -2,47 +2,44 @@
...
 properties([
     pipelineTriggers([cron('10 21 * * *')]),
-
     parameters([
-
-        string(name: 'URL_TO_TEST', defaultValue: 'https://idam-web-public-idam-preview.service.core-compute-idam-preview.internal', description: 'The URL you want to run these tests against'),
-        string(name: 'API_URL_TO_TEST', defaultValue: 'https://idam-api-idam-preview.service.core-compute-idam-preview.internal', description: 'The API URL you want to run these tests against '),
+        string(name: 'URL_TO_TEST', defaultValue: 'https://idam-web-public-idam-aat.service.core-compute-idam-aat.internal', description: 'The URL you want to run these tests against'),
+        string(name: 'API_URL_TO_TEST', defaultValue: 'https://idam-api-idam-aat.service.core-compute-idam-aat.internal', description: 'The API URL you want to run these tests against '),
+        string(name: 'SAUCE_KEY', defaultValue: '101a14e3-0458-4927-a968-129adbf18152', description: 'Sauce key'),
```
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: as previously mentioned its should run on AAT which is master

File: Jenkinsfile_nightly [hunk 15]
```diff
@@ -2,47 +2,44 @@
...
 def type = "java"
-
 def product = "idam"
-
-def component = "web-public"
-
+def component = "idam-web-public"
 def secrets = [
-    'idam-idam-preview': [
+    'idam-idam-aat':[
```
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: > @sudhasane This PR need refinement. It should not have been branched from such an old branch really and there are many quality issues. I'd like to see a link to a passing build using the nightly Jenkinsfile but before that, the functionality of the nightly pipeline is not as desired. This pipeline file will only all tests against one environment despite the parameters for the test urls. Please address these issues and ensure that the pipeline can target any environment and defaults to aat.

We run nightly tests only on AAT thats the requirement from hmcts

[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: > Ideally this should be dynamic based on the environment you are targeting. This only facilitates testing in one environments.

as previously mentioned its should run on AAT which is master

File: src/test/js/config/test_data.js (hunk 7 above)
Comment (sudhasane) [reply src/test/js/config/test_data.js]: This is not part of my change but still I will check

**Comment by sudhasane** [comment]
> @sudhasane This PR need refinement. It should not have been branched from such an old branch really and there are many quality issues. I'd like to see a link to a passing build using the nightly Jenkinsfile but before that, the functionality of the nightly pipeline is not as desired. This pipeline file will only all tests against one environment despite the parameters for the test urls. Please address these issues and ensure that the pipeline can target any environment and defaults to aat.

we run nightly tests only on AAT thats the requirement from hmcts

File: Jenkinsfile_nightly (hunk 14 above)
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: do we have vault keys for URL_TO_TEST and API_URL, if you can provide key I can add that in the file.

File: Jenkinsfile_nightly (hunk 13 above)
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: I Just checked with HMCTS QA lead , sauce key and user name can be removed from nightly-jenkins file as they were taken care by jenkins while running

File: Jenkinsfile_nightly (hunk 15 above)
[the remark this answers is not in the document]
Comment (sudhasane) [reply Jenkinsfile_nightly]: https://github.com/hmcts/cnp-jenkins-config/blob/master/jobdsl/organisations.groovy#L82
